File: src/main/kotlin/com/wxm/demo/util/PageUtil.java
package com.wxm.demo.util;

import java.io.Serializable;
import java.util.List;

/**
 * @Title: PageUtils.java
 * @Description: TODO
 * @Author JPFee
 * @Date 2017/11/169:30
 * @Annotation 分页工具类
 */
public class PageUtil implements Serializable {
    private static final long serialVersionUID = 1L;
    // 总记录数
    private int total;
    // 列表数据
    private List<?> rows;

    /**
     * 分页
     *
     * @param list
     *            列表数据
     */
    public PageUtil(List<?> list, int total) {
        this.rows = list;
        this.total = total;
    }

    public PageUtil(List<?> list)   {
        this.rows = list;
        this.total = list.size();
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public List<?> getRows() {
        return rows;
    }

    public void setRows(List<?> rows) {
        this.rows = rows;
    }

}

File: src/main/kotlin/com/wxm/demo/common/util/PageUtil.kt
package com.wxm.demo.common.util

import java.io.Serializable

/**
 * @Title: PageUtils.java
 * @Description: TODO
 * @Author JPFee
 * @Date 2017/11/169:30
 * @Annotation 分页工具类
 */
class PageUtil : Serializable {
    // 总记录数
    var total: Int = 0
    // 列表数据
    var rows: List<*>? = null

    /**
     * 分页
     *
     * @param list
     * 列表数据
     */
    constructor(list: List<*>, total: Int) {
        this.rows = list
        this.total = total
    }

    constructor(list: List<*>) {
        this.rows = list
        this.total = list.size
    }

    companion object {
        private const val serialVersionUID = 1L
    }

}
